src/ark/utils/data_utils.py

```diff
@@ -534,16 +534,22 @@ def stitch_images_by_shape(data_dir, stitched_dir, img_sub_folder=None, channels
                                   valid_channels=io_utils.remove_file_extensions(channel_imgs))
 
     file_ext = channel_imgs[0].split('.')[1]
-    expected_fovs, num_rows, num_cols = load_utils.get_tiled_fov_names(fovs, return_dims=True)
+    expected_tiles = load_utils.get_tiled_fov_names(fovs, return_dims=True)
 
     # save new images to the stitched_images, one channel at a time
-    os.makedirs(stitched_dir)
     for chan in channels:
-        image_data = load_utils.load_tiled_img_data(data_dir, fovs, expected_fovs, chan,
-                                                    single_dir=any([segmentation, clustering]),
-                                                    file_ext=file_ext,
-                                                    img_sub_folder=img_sub_folder)
-        stitched_data = data_utils.stitch_images(image_data, num_cols)
-        current_img = stitched_data.loc['stitched_image', :, :, chan].values
-        image_utils.save_image(os.path.join(stitched_dir, chan + '_stitched.' + file_ext),
-                               current_img)
+        for tile in expected_tiles:
```

Comment:
We can remove the nested loop with [`itertools.product`](https://docs.python.org/3/library/itertools.html#itertools.product).